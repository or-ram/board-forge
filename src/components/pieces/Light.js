import React, { PureComponent } from 'react';
import * as P from 'prop-types';

export default class Light extends PureComponent {
  static propTypes = {
    crowned: P.bool,
    height: P.number,
    width: P.number,
  };

  static defaultProps = {
    crowned: false,
    height: 70,
    width: 70,
  };

  render () {
    const { crowned, height, width } = this.props;

    return (
      <div className="man">
        <svg width={width} height={height} viewBox="0 0 80 80">
          <circle cx="40" cy="40" r="38" stroke="black" strokeWidth="4" fill="white" />
          <circle cx="40" cy="40" r="22" stroke="black" strokeWidth="4" fill="white" />
          {
            crowned &&
            <circle cx="40" cy="40" r="14" fill="black" />
          }
        </svg>
      </div>
    );
  }
}
