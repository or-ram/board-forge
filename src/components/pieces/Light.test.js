import React from 'react';
import renderer from 'react-test-renderer';

import Light from './Light';

describe('Light Test', () => {
  const component = renderer.create(<Light />);
  let tree = null;

  beforeEach(() => {
    tree = component.toJSON();
  });

  test('Correctly rendered', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Have only 1 children', () => {
    expect(tree.children).toHaveLength(1);
  });

  test('Have 2 childrens', () => {
    const tmpTree = renderer.create(<Light crowned={true} />).toJSON();

    expect(tmpTree.children[0].children).toHaveLength(3);
  });
});
