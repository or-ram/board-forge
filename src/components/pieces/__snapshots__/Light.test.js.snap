// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Light Test Correctly rendered 1`] = `
<div
  className="man"
>
  <svg
    height={70}
    viewBox="0 0 80 80"
    width={70}
  >
    <circle
      cx="40"
      cy="40"
      fill="white"
      r="38"
      stroke="black"
      strokeWidth="4"
    />
    <circle
      cx="40"
      cy="40"
      fill="white"
      r="22"
      stroke="black"
      strokeWidth="4"
    />
  </svg>
</div>
`;
